doc/1.7/dev_guide/lua_style_guide.rst

```diff
@@ -0,0 +1,465 @@
+                   Avoid global variable
+===========================================================
+
+Avoid global variables. Strongly avoid. Strongly-strongly avoid them. If you
+have and exceptional case, use ``_G`` variable to set it, add prefix or add
+table instead of prefix:
+
+.. code-block:: lua
+
+    function bad_global_example()
+    end -- very, very bad
+
+    function good_local_example()
+    end
+    _G.modulename_good_local_example = good_local_example -- good
+    _G.modulename = {}
+    _G.modulename.good_local_example = good_local_example -- better
+
+Always use prefix to avoid interception of names
+
+===========================================================
+                           Naming
+===========================================================
+
+* names of variables/"objects" and "methods"/functions: snake_case
+* names of "classes": CamelCase
+* private variables/methods (properties in the future) of object starts with
+  underscores ``<object>._<name>``. Avoid using of
+  ``local function private_methods(self) end``
+* boolean - naming ``is_<...>``, ``isnt_<...>``, ``has_``, ``hasnt_`` is a good style.
+* for "very local" variables:
+  - ``t`` is for tables
+  - ``i``, ``j`` are for indexing
+  - ``n`` is for counting
+  - ``k``, ``v`` is what you get out of ``pairs()`` (are acceptable, ``_`` if unused)
+  - ``i``, ``v`` is what you get out of ``ipairs()`` (are acceptable, ``_`` if unused)
+  - ``k``/``key`` is for table keys
+  - ``v``/``val``/``value`` is for values that are passed around
+  - ``x``/``y``/``z`` is for generic math quantities
+  - ``s``/``str``/``string`` is for strings
+  - ``c`` is for 1-char strings
+  - ``f``/``func``/``cb`` are for functions
+  - ``status, <return_name>`` is what you get out of pcall
+  - ``buf, sz`` is a (buffer, size) pair
+  - ``<name>_p`` is for pointers
+  - ``t0``.. is for timestamps
+  - ``err`` is for errors
+* abbrevations are acceptable if they're unambigous and if you'll document (or
+  they're too common) them.
+* global variables are written with ALL_CAPS. If it's some system variable, then
+  they're using underscore to define it (``_G``/``_VERSION``/..)
+* module naming snake_case (avoid underscores and dashes) - 'luasql', instead of
+  'Lua-SQL'
+* ``*_mt`` and ``*_methods`` defines metatable and methods table
+
+===========================================================
+                    Idioms and patterns
+===========================================================
+
+Always use round brackets in call of functions except multiple cases:
+
+* ``*.cfg{ }`` functions (``box.cfg``/``memcached.cfg``/..)
+* ``ffi.cdef[[ ]]`` function
+
+Avoid these kind of constructions:
+
+* <func>'<name>' (strongly avoid require'..')
+* ``function object:method() end`` (use ``functon object.method(self) end`` instead)
+* use comma separator in tables, instead ``;`` or anyone else
+* put comma at the end of line
```

Comment:
I mean semicolon here, thanks. Semicolon may be used as 'visual separator in Lua'